Content.Server/Cloning/CloningSystem.cs

```diff
@@ -256,6 +263,19 @@ public bool TryCloning(EntityUid uid, EntityUid bodyToClone, Mind.Mind mind, Clo
             return true;
         }
 
+        private void OnZombieCloning(EntityUid uid, ZombieComponent zombiecomp, ref CloningEvent args)
+        {
+            foreach (var (layer, info) in zombiecomp.BeforeZombifiedCustomBaseLayers)
+            {
+                _humanoidSystem.SetBaseLayerColor(args.Target, layer, info.Color);
+                _humanoidSystem.SetBaseLayerId(args.Target, layer, info.ID);
+            }
+            _humanoidSystem.SetSkinColor(args.Target, zombiecomp.BeforeZombifiedSkinColor);
+
+            MetaData(args.Target).EntityName = zombiecomp.BeforeZombifiedEntityName;
+            args.NameHandled = true;
+        }
+
```

Comment:
This should be on zombiesystem and should be a generic unzombify rather than hardcoding this.

Reply:
I just moved this to zombiesystem. The cloning is not exactly an unzombify as we are working on 2 different entities/mob and the new one doesn't have any zombiecomponent.